.github/workflows/copilot-build-memorypipeline.yml

```diff
@@ -0,0 +1,72 @@
+﻿name: memorypipeline-build
+
+on:
+  pull_request:
+    branches: ["main"]
+    paths:
+      - "memorypipeline/**"
+  workflow_call:
+    outputs:
+      artifact:
+        description: "The name of the uploaded memory pipeline artifact."
+        value: ${{jobs.memory-pipeline.outputs.artifact}}
+
+permissions:
+  contents: read
+
+jobs:
+  memory-pipeline:
+    runs-on: windows-latest
+
+    env:
+      NUGET_CERT_REVOCATION_MODE: offline
+
+    outputs:
+      artifact: ${{steps.artifactoutput.outputs.artifactname}}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          clean: true
+
+      - name: Install GitVersion
+        uses: gittools/actions/gitversion/setup@v0
+        with:
+          versionSpec: "5.x"
+
+      - name: Determine version
+        id: gitversion
+        uses: gittools/actions/gitversion/execute@v0
+
+      - name: Set version tag
+        id: versiontag
+        run: |
+          $VERSION_TAG = "${{ steps.gitversion.outputs.Major }}."
```

Comment:
This is probably going to crap out unless you specify a fetch depth of 0 (unlimited) in the step above:

      - uses: actions/checkout@v3
        with:
          clean: true
          fetch-depth: 0